### PR TITLE
[improve][broker] Make some methods of `ClusterBase` pure async.

### PR DESCRIPTION
**Pull request by mattisonchao**
### Motivation

See PIP #14365  and change tracker #15043. 

### Modifications

- Make `ClusterBase` `getBrokerWithNamespaceIsolationPolicy ` / `setNamespaceIsolationPolicy` to pure async.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

`AdminTest#clusters` already cover this change.

### Documentation

- [x] `no-need-doc` 
(Please explain why)

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java [hunk 1]
```diff
@@ -64,12 +65,13 @@
 import org.apache.pulsar.common.policies.impl.NamespaceIsolationPolicies;
 import org.apache.pulsar.common.policies.impl.NamespaceIsolationPolicyImpl;
 import org.apache.pulsar.common.util.FutureUtil;
-import org.apache.pulsar.common.util.ObjectMapperFactory;
 import org.apache.pulsar.metadata.api.MetadataStoreException;
 import org.apache.pulsar.metadata.api.MetadataStoreException.NotFoundException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import static javax.ws.rs.core.Response.Status.PRECONDITION_FAILED;
```
Comment (liudezhi2098) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java:73]: check checkstyle

Comment (mattisonchao) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java]: fixed, thanks.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java [hunk 2]
```diff
@@ -522,8 +523,9 @@ private BrokerNamespaceIsolationData internalGetBrokerNsIsolationData(
             NamespaceIsolationPolicyImpl nsPolicyImpl = new NamespaceIsolationPolicyImpl(policyData);
             if (nsPolicyImpl.isPrimaryBroker(broker) || nsPolicyImpl.isSecondaryBroker(broker)) {
                 namespaceRegexes.addAll(policyData.getNamespaces());
-                brokerIsolationData.primary(nsPolicyImpl.isPrimaryBroker(broker));
-                brokerIsolationData.policyName(name);
+                if (nsPolicyImpl.isPrimaryBroker(broker)) {
+                    brokerIsolationData.primary(true);
+                }
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java:528]: No need to change.

File: pulsar-common/src/main/java/org/apache/pulsar/common/util/FutureUtil.java [hunk 3]
```diff
@@ -47,6 +50,18 @@ public static CompletableFuture<Void> waitForAll(Collection<? extends Completabl
...
+    public static <T> CompletableFuture<List<T>> waitForAll(Stream<CompletableFuture<List<T>>> futures) {
+        return futures.reduce(CompletableFuture.completedFuture(new ArrayList<>()),
+                (pre, curr) -> pre.thenCompose(preV -> curr.thenApply(currV -> {
+                    preV.addAll(currV);
+                    return preV;
+                })));
+    }
+
+    public static <T> CompletableFuture<List<T>> waitForAll(List<CompletableFuture<List<T>>> futures) {
+        return waitForAll(futures.stream());
```
Comment (Technoboy-) [inline pulsar-common/src/main/java/org/apache/pulsar/common/util/FutureUtil.java:62]: And here.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java (hunk 2 above)
Comment (mattisonchao) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java:528]: Yes, It's an issue caused by rebase operation, I will fix it.

File: pulsar-common/src/main/java/org/apache/pulsar/common/util/FutureUtil.java (hunk 3 above)
Comment (mattisonchao) [reply pulsar-common/src/main/java/org/apache/pulsar/common/util/FutureUtil.java:62]: Yes, I will fix it and mark it resolved.

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**